api/test/wfl/integration/modules/covid_test.clj

```diff
@@ -85,78 +81,102 @@
+                         :methodConfiguration        "mc-namespace/mc-name"
+                         :methodConfigurationVersion 1
+                         :fromSource                 "importSnapshot"
+                         :skipValidation             true}
+                        (covid/create-executor! tx (rand-int 1000000))
+                        (zipmap [:executor_type :executor_items])
+                        (covid/load-executor! tx)))]
+    (letfn [(verify-record-against-workflow [record workflow idx]
               (is (= idx (:id record)))
               (is (= (:status workflow) (:workflow_status record)))
               (is (= (:workflowId workflow) (:workflow_id record))))]
-        (with-redefs-fn
-          {#'rawls/create-snapshot-reference   mock-rawls-create-snapshot-reference
-           #'covid/create-submission!          mock-create-submission}
-          #(covid/update-executor! source executor))
-        (is (-> source :queue empty?) "The snapshot was not consumed.")
-        (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
-          (let [[running-record succeeded-record & _ :as records]
-                (->> executor :details (postgres/get-table tx))]
-            (is (== 2 (count records))
-                "The workflows were not written to the database")
-            (is (every? #(= snapshot-reference-id (:snapshot_reference_id %)) records))
-            (is (every? #(= submission-id (:rawls_submission_id %)) records))
-            (is (every? #(nil? (:consumed %)) records))
-            (verify-record-against-workflow running-record running-workflow 1)
-            (verify-record-against-workflow succeeded-record succeeded-workflow 2)))))))
+      (with-redefs-fn
+        {#'rawls/create-snapshot-reference   mock-rawls-create-snapshot-reference
+         #'covid/create-submission!          mock-create-submission}
+        #(covid/update-executor! source executor))
+      (is (-> source :queue empty?) "The snapshot was not consumed.")
+      (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+        (let [[running-record succeeded-record & _ :as records]
+              (->> executor :details (postgres/get-table tx))]
+          (is (== 2 (count records))
+              "The workflows were not written to the database")
+          (is (every? #(= snapshot-reference-id (:snapshot_reference_id %)) records))
+          (is (every? #(= submission-id (:rawls_submission_id %)) records))
+          (is (every? #(nil? (:consumed %)) records))
+          (verify-record-against-workflow running-record running-workflow 1)
+          (verify-record-against-workflow succeeded-record succeeded-workflow 2))))))
+
+(deftest test-peek-terra-executor-queue
+  (let [succeeded? #{"Succeeded"}
+        source     (make-queue-from-list [{:name "test-snapshot-name"
+                                           :id   (str (UUID/randomUUID))}])
+        executor   (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
```

Comment:
Thanks for adding this test as well.  Nit: might be good to break out this assignment into a helper since we use it in the above test.

Reply:
Sure. Nit: it's not an assignment, it's a binding :-p